superset/models/core.py

```diff
@@ -767,6 +767,15 @@ def get_sqla_engine(self, schema=None, nullpool=True, user_name=None):
         if DB_CONNECTION_MUTATOR:
             url, params = DB_CONNECTION_MUTATOR(
                 url, params, effective_username, security_manager)
+
+        if (sqllab and self.db_engine_spec.custom_sqllab_cursor_params()):
+            url = '{}?source={}'.format(
```

Comment:
I tried passing source into the params object instead of this url appending approach but I got the error: 
`TypeError: Invalid argument(s) 'source' sent to create_engine(), using configuration PrestoDialect/NullPool/Engine.  Please check that the keyword arguments are appropriate for this combination of components.
`